runners/cli/src/main/kotlin/cli/main.kt

```diff
@@ -45,41 +46,90 @@ class DokkaArguments {
     var jdkVersion: Int = 6
 }
 
-fun main(args: Array<String>) {
-    val arguments = DokkaArguments()
-    val freeArgs: List<String> = Args.parse(arguments, args, false) ?: listOf()
-    val sources = if (arguments.src.isNotEmpty()) arguments.src.split(File.pathSeparatorChar).toList() + freeArgs else freeArgs
-    val samples = if (arguments.samples.isNotEmpty()) arguments.samples.split(File.pathSeparatorChar).toList() else listOf()
-    val includes = if (arguments.include.isNotEmpty()) arguments.include.split(File.pathSeparatorChar).toList() else listOf()
-
-    val sourceLinks = if (arguments.srcLink.isNotEmpty() && arguments.srcLink.contains("="))
-        listOf(parseSourceLinkDefinition(arguments.srcLink))
-    else {
-        if (arguments.srcLink.isNotEmpty()) {
-            println("Warning: Invalid -srcLink syntax. Expected: <path>=<url>[#lineSuffix]. No source links will be generated.")
+
+object MainKt {
+
+    @JvmStatic
+    fun entry(args: Array<String>) {
+        val arguments = DokkaArguments()
+        val freeArgs: List<String> = Args.parse(arguments, args, false) ?: listOf()
+        val sources = if (arguments.src.isNotEmpty()) arguments.src.split(File.pathSeparatorChar).toList() + freeArgs else freeArgs
+        val samples = if (arguments.samples.isNotEmpty()) arguments.samples.split(File.pathSeparatorChar).toList() else listOf()
+        val includes = if (arguments.include.isNotEmpty()) arguments.include.split(File.pathSeparatorChar).toList() else listOf()
+
+        val sourceLinks = if (arguments.srcLink.isNotEmpty() && arguments.srcLink.contains("="))
+            listOf(parseSourceLinkDefinition(arguments.srcLink))
+        else {
+            if (arguments.srcLink.isNotEmpty()) {
+                println("Warning: Invalid -srcLink syntax. Expected: <path>=<url>[#lineSuffix]. No source links will be generated.")
+            }
+            listOf()
+        }
+
+        val classPath = arguments.classpath.split(File.pathSeparatorChar).toList()
+
+        val documentationOptions = DocumentationOptions(
+                arguments.outputDir.let { if (it.endsWith('/')) it else it + '/' },
+                arguments.outputFormat,
+                skipDeprecated = arguments.nodeprecated,
+                sourceLinks = sourceLinks
+        )
+
+        val generator = DokkaGenerator(
+                DokkaConsoleLogger,
+                classPath,
+                sources,
+                samples,
+                includes,
+                arguments.moduleName,
+                documentationOptions)
+
+        generator.generate()
+        DokkaConsoleLogger.report()
+    }
+
+    fun findToolsJar(): File {
+        val javaHome = System.getProperty("java.home")
+        val default = File(javaHome, "../lib/tools.jar")
+        val mac = File(javaHome, "../Classes/classes.jar")
```

Comment:
Is this actually needed? I think this is the same jar where all the rest of JRE classes are located, so if it's not on the classpath, we won't get to this point.